src/sample.rs

```diff
@@ -27,173 +29,285 @@ use std::str::FromStr;
+                std::io::stdout()
+                    .flush()
+                    .expect("error: Failed to flush stdout");
+            }
+        }
+    }
+}
+
+fn cli() -> Command {
+    // strip out usage
+    const PARSER_TEMPLATE: &str = "\
+        {all-args}
+    ";
+    // strip out name/version
+    const APPLET_TEMPLATE: &str = "\
+        {about-with-newline}\n\
+        {usage-heading}\n    {usage}\n\
+        \n\
+        {all-args}{after-help}\
+    ";
+
+    Command::new("Nostr Client REPL")
+        .multicall(true)
+        .arg_required_else_help(true)
+        .subcommand_required(true)
+        .help_template(PARSER_TEMPLATE)
+        .subcommand(
+            Command::new("sendtextnote")
+                .args([arg!(-c --content <CONTENT> "Send a text note to the relay")])
+                .help_template(APPLET_TEMPLATE)
+                .about("Send a text note to the relay"),
+        )
+        .subcommand(
+            Command::new("setmetadata")
+                .args([
+                    arg!(-u --username <USERNAME> "The client's username"),
+                    arg!(-a --about <ABOUT> "The client's about string"),
+                    arg!(-p --picture <PICTURE> "The client's picture"),
+                ])
+                .help_template(APPLET_TEMPLATE)
+                .about("Set the client's metadata"),
+        )
+        .subcommand(
+            Command::new("recommendserver")
+                .args([arg!(-u --urlrelay <URLRELAY> "The url string of the server to recommend")])
+                .help_template(APPLET_TEMPLATE)
+                .about("Recommend a server to the relay"),
+        )
+        .subcommand(
+            Command::new("opensubscription")
+                .args([
+                    arg!(-i --subscriptionid <SUBSCRIPTIONID> "The subscription id"),
+                    arg!(-k --kinds <KINDS> "The kinds of events to subscribe to"),
+                    arg!(-s --since <SINCE> "The timestamp to start the subscription"),
+                    arg!(-u --until <UNTIL> "The timestamp to end the subscription"),
+                ])
+                .help_template(APPLET_TEMPLATE)
+                .about("Open a subscription to the relay"),
+        )
+        .subcommand(
+            Command::new("closesubscription")
+                .args([arg!(-i --subscriptionid <SUBSCRIPTIONID> "The subscription id")])
+                .help_template(APPLET_TEMPLATE)
+                .about("Close a subscription to the relay"),
+        )
+        .subcommand(
+            Command::new("shutdown")
+                .help_template(APPLET_TEMPLATE)
+                .about("Shutdown the REPL"),
+        )
+}
+
+fn respond(
+    line: &str,
+    tx: &futures_channel::mpsc::UnboundedSender<Message>,
+    client_keys: &Keys,
+) -> Result<bool, String> {
+    let args = shlex::split(&line).expect("error: Failed to parse command");
```

Comment:
I am not completely aware of the edge cases that this currently covers, but we could avoid the need for this extra dependency by keep using `.split_whitespace()`. I've done a few tests without noticing any issues.
```suggestion
    let args = line.split_whitespace().collect::<Vec<&str>>();
```

Reply:
Makes sense